.github/workflows/generate-code-coverage.yaml

```diff
@@ -0,0 +1,51 @@
+name: Generate Code Coverage
+on: [pull_request]
+  
+jobs:
+  cover-base:
+    name: Generate Base Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout base
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+
+      - name: Set up go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20.2'
```

Comment:
Maybe overkill, but can we avoid hardcoding with an env var? https://stackoverflow.com/questions/74021760/how-to-use-environment-variables-on-github-actions-without-hard-coding-them-is

Reply:
I'm not sure I understand what you're suggesting? Can you clarify?

Reply:
We currently hardcode the go version for other workflows as well but its a good idea to grab it from go mod using `go-version-file` : https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file.